### DISCUSSION TITLE
4.2a4 Bug Report

### DISCUSSION THREAD

**Discussion by ProfFan** [borglab/gtsam-manylinux-build#29] [Q&A]
Paste your install log and `pip debug --verbose` below:

**Comment by jmao44** [comment]
- Platform: MacBook Pro, Intel
- OS: macOS Monterey 12.1

Error below is thrown when trying to install gtsam 4.2a4 via wheel downloaded [here](https://github.com/borglab/gtsam-manylinux-build/actions/runs/1758999682).
`ERROR: gtsam-4.2a4-cp38-cp38-macosx_11_0_x86_64.whl is not a supported wheel on this platform.`

Logs of `pip debug --verbose` is attached below
[pip_debug_gtsam42a4_py38_macos12_intel.txt](https://github.com/borglab/gtsam-manylinux-build/files/7960480/pip_debug_gtsam42a4_py38_macos12_intel.txt)

**Comment by dellaert** [comment]
Yes, it's maddening that on MacOS 12, macOS 11 wheels are not recognized as valid. Is this because minoconda needs to be updated or something?

**Reply by ProfFan** [reply]
I think miniconda has its own "fixed" environment where they fix the `MACOS_DEPLOYMENT_TARGET` to 10.15 or something so they don't need to build separate binaries? Saves cost for the company...

**Comment by dellaert** [comment]
OK, then let's get the 10.15 wheels built correctly as soon as possible

**Reply by ProfFan** [reply]
They are built now, we need to try it

**Reply by jmao44** [reply]
I just installed gtsam via the 10.15 wheel on macOS 12. It's giving me the following error when I run `import gtsam`.
```
>>> import gtsam
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/maojy/opt/miniconda3/envs/py38_test/lib/python3.8/site-packages/gtsam/__init__.py", line 7, in <module>
    from gtsam import gtsam, utils
ImportError: dlopen(/Users/maojy/opt/miniconda3/envs/py38_test/lib/python3.8/site-packages/gtsam/gtsam.cpython-38-darwin.so, 0x0002): Library not loaded: @loader_path/../../../gtsam.dylibs/libgtsam.4.2.0.dylib
  Referenced from: /Users/maojy/opt/miniconda3/envs/py38_test/lib/python3.8/site-packages/gtsam/gtsam.cpython-38-darwin.so
  Reason: tried: '/Users/maojy/opt/miniconda3/envs/py38_test/lib/python3.8/site-packages/gtsam/../../../gtsam.dylibs/libgtsam.4.2.0.dylib' (no such file), '/usr/local/lib/libgtsam.4.2.0.dylib' (no such file), '/usr/lib/libgtsam.4.2.0.dylib' (no such file)
```
